src/android/src/main/java/io/callstack/react/fbads/InterstitialAdManager.java

```diff
@@ -0,0 +1,93 @@
+package io.callstack.react.fbads;
+
+import com.facebook.ads.Ad;
+import com.facebook.ads.AdError;
+import com.facebook.ads.InterstitialAd;
+import com.facebook.ads.InterstitialAdListener;
+import com.facebook.react.bridge.LifecycleEventListener;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+
+public class InterstitialAdManager extends ReactContextBaseJavaModule implements InterstitialAdListener, LifecycleEventListener {
+
+  private Promise mPromise;
+  private boolean mDidClick = false;
+  private InterstitialAd mInterstitial;
+
+  public InterstitialAdManager(ReactApplicationContext reactContext) {
+    super(reactContext);
+    reactContext.addLifecycleEventListener(this);
+  }
+
+  @ReactMethod
+  public void showAd(String placementId, Promise p) {
+    if (mPromise != null) {
+      return;
+    }
+    ReactApplicationContext reactContext = this.getReactApplicationContext();
+
+    mPromise = p;
+    mInterstitial = new InterstitialAd(reactContext, placementId);
+    mInterstitial.setAdListener(this);
+    mInterstitial.loadAd();
+  }
+
+  @Override
+  public String getName() {
+    return "CTKInterstitialAdManager";
+  }
+
+  @Override
+  public void onError(Ad ad, AdError adError) {
+    mPromise.reject("E_FAILED_TO_LOAD", adError.getErrorMessage());
+    cleanUp();
+  }
+
+  @Override
+  public void onAdLoaded(Ad ad) {
+    if (ad == mInterstitial) {
+      mInterstitial.show();
+    }
+  }
+
+  @Override
+  public void onAdClicked(Ad ad) {
+    mDidClick = true;
+  }
+
+  @Override
+  public void onInterstitialDismissed(Ad ad) {
+    mPromise.resolve(mDidClick);
+    cleanUp();
+  }
+
+  @Override
+  public void onInterstitialDisplayed(Ad ad) {
+
+  }
+
+  private void cleanUp() {
+    mPromise = null;
+    mDidClick = false;
```

Comment:
I did some checks and it looks like `mInterstital.destroy()` does not set `mInterstital` to `null`, actually, if I don't set it to `null` manually and I want to reuse it on `showAd` again even after closing whole app I can... So I'd suggest to create a function that does this:

```
if (mInterstitial != null) {
  mInterstitial.destroy();
  mInterstitial = null;
}
```

And call it on `cleanUp` and `onHostDestroy`.